apps/angular/58-content-projection-condition/src/app/card.component.ts

```diff
@@ -1,25 +1,50 @@
-import { ChangeDetectionStrategy, Component, input } from '@angular/core';
+import { NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  contentChild,
+  Directive,
+  input,
+  TemplateRef,
+} from '@angular/core';
+
+@Directive({
+  selector: '[appCardTitle]',
+})
+export class CardTitleDirective {}
+
+@Directive({
+  selector: '[appCardMessage]',
+})
+export class CardMessageDirective {}
```

Comment:
nice, using a directive to avoid magic string is the cherry on top. Nice solution